apis/v1alpha1/opentelemetrycollector_types.go

```diff
@@ -177,6 +177,10 @@ type OpenTelemetryTargetAllocator struct {
 	// All CR instances which the ServiceAccount has access to will be retrieved. This includes other namespaces.
 	// +optional
 	PrometheusCR OpenTelemetryTargetAllocatorPrometheusCR `json:"prometheusCR,omitempty"`
+	// PodMonitors to be selected for target discovery.
+	PodMonitorSelector *map[string]string `json:"podMonitorSelector,omitempty"`
+	// ServiceMonitors to be selected for target discovery.
```

Comment:
We should reference what the options are for the fields e.g. the link to the API reference or better an user guide. https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.24/#labelselector-v1-meta

Reply:
I originally used the metav1.LabelSelector here, but it wasn't serializing & deserializing to YAML correctly, so i did what we do for the collector label selector. I can link these docs though.